server/text_generation_server/models/causal_lm.py

```diff
@@ -630,6 +642,52 @@ def __init__(
             self.hb_profer_started = False
         self.step = 0
 
+    def setup_env(self):
+        import sys
+        assert "habana_frameworks" not in sys.modules
+
+        self.is_quantization_enabled = os.getenv("QUANT_CONFIG", "") != ""
+        if self.is_quantization_enabled:
+            os.environ.setdefault("ENABLE_EXPERIMENTAL_FLAGS", "true")
+            os.environ.setdefault("USE_DEFAULT_QUANT_PARAM", "true")
+            os.environ.setdefault("UPDATE_GRAPH_OUTPUT_MME", "false")
+            os.environ.setdefault("ENABLE_CALC_DYNAMIC_RANGE", "false")
+            os.environ.setdefault(
+                "UPDATE_MME_OUTPUT_PRECISION_FILTER", "v_proj,matmul_av")
+            os.environ.setdefault("EXPERIMENTAL_WEIGHT_SHARING", "FALSE")
+            import habana_frameworks.torch.core as htcore
+            htcore.hpu_set_env()
+
+    def setup_quantization(self, model):
```

Comment:
do we need to have 2 setup functions? "setup_env" and "setup_quantization". It seems that both of them are here for the same reason

Reply:
setup_env is called before habana_frameworks is loaded, it's needed to setup necessary environment variables for synapse.

Reply:
they are separated to distinguish two kinds of initialization:
- setup_env - no model modifications, just setting necessary env variables and enabling inference for the whole habana_frameworks
- setup_quantization - specific model modifications for quantization